File: app/src/main/java/com/dpis/module/QuickConfigEditorSession.java
package com.dpis.module;
import com.dpis.module.appconfig.EditorDraft;

import com.dpis.module.applist.AppListItem;

/**
 * In-memory Quick Config editing session retained only across Activity configuration changes.
 * Closing the Activity still discards the draft instead of turning it into persisted config.
 */
final class QuickConfigEditorSession {

    final AppListItem item;
    final EditorDraft draft;
    final EditorDraft savedDraft;
    final ConfigEditorDestination destination;

    QuickConfigEditorSession(
            AppListItem item,
            EditorDraft draft,
            EditorDraft savedDraft,
            ConfigEditorDestination destination
    ) {
        this.item = item;
        this.draft = draft;
        this.savedDraft = savedDraft != null ? savedDraft : draft;
        this.destination = destination != null ? destination : ConfigEditorDestination.MAIN;
    }
}

File: app/src/main/java/com/dpis/module/quickconfig/QuickConfigEditorSession.kt
package com.dpis.module.quickconfig

import com.dpis.module.appconfig.AppConfigEditorSession
import com.dpis.module.appconfig.EditorDraft
import com.dpis.module.applist.AppListItem
import com.dpis.module.ui.ConfigEditorDestination

/**
 * In-memory Quick Config editing session retained only across Activity configuration changes.
 * Closing the Activity still discards the draft instead of turning it into persisted config.
 */
internal class QuickConfigEditorSession @JvmOverloads constructor(
    @JvmField val item: AppListItem?,
    @JvmField val draft: EditorDraft,
    savedDraft: EditorDraft?,
    destination: ConfigEditorDestination?,
    @JvmField val editorSession: AppConfigEditorSession? = null,
) {
    @JvmField val savedDraft: EditorDraft = savedDraft ?: draft
    @JvmField val destination: ConfigEditorDestination = destination ?: ConfigEditorDestination.MAIN
}
